src/main/java/org/sagebionetworks/web/client/widget/doi/DoiWidgetV2ViewImpl.java

```diff
@@ -0,0 +1,106 @@
+package org.sagebionetworks.web.client.widget.doi;
+
+import org.gwtbootstrap3.client.ui.Icon;
+import org.gwtbootstrap3.client.ui.constants.IconType;
+import org.gwtbootstrap3.client.ui.html.Span;
+import org.sagebionetworks.web.client.DisplayUtils;
+import org.sagebionetworks.web.client.GlobalApplicationState;
+import org.sagebionetworks.web.client.security.AuthenticationController;
+
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiField;
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwt.user.client.ui.TextBox;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+
+public class DoiWidgetV2ViewImpl implements DoiWidgetV2View {
+	
+	GlobalApplicationState globalApplicationState;
+	AuthenticationController authenticationController;
+	@UiField
+	Icon clipboardIcon;
+	@UiField
+	TextBox doi;
+	@UiField
+	Span doiLabel;
+	@UiField
+	Span synAlertContainer;
+
+	Widget widget;
+	
+	public interface Binder extends UiBinder<Widget, DoiWidgetV2ViewImpl> {}
+	
+	@Inject
+	public DoiWidgetV2ViewImpl(GlobalApplicationState globalApplicationState,
+							   AuthenticationController authenticationController,
+							   Binder uiBinder) {
+		this.globalApplicationState = globalApplicationState;
+		this.authenticationController = authenticationController;
+		widget = uiBinder.createAndBindUi(this);
+
+		doi.addClickHandler(event -> copyDoiToClipboard());
+
+		clipboardIcon.addClickHandler(event -> copyDoiToClipboard());
+	}
+
+	/**
+	 * See https://stackoverflow.com/questions/1317052/how-to-copy-to-clipboard-with-gwt
+	 */
+	private void copyDoiToClipboard() {
+		doi.setFocus(true);
+		doi.selectAll();
+		boolean success = copyToClipboard();
+		if (success) {
+			clipboardIcon.setType(IconType.CHECK);
```

Comment:
Do not change icon (able to copy multiple times).   Instead, showInfo "Successfully copied to the clipboard"